src/os2datascanner/engine/utils.py

```diff
@@ -53,3 +59,37 @@ def get_data(file_name: str, uppercase=True) -> typing.FrozenSet[str]:
         for line in text.splitlines()
         if line
     )
+
+
+@contextmanager
+def prometheus_session(name, **kwargs):
+    while True:
+        # Find a free port to serve Prometheus metrics over...
+        try:
+            port = randrange(45000, 55000)
+            start_http_server(port)
+            break
+        except OSError as ex:
+            if ex.errno == errno.EADDRINUSE:
+                continue
+            else:
+                raise
+
+    # ... advertise this port, and this service, to Prometheus...
+    print(__file__)
+    advertisement_path = os.path.dirname(__file__) + "/../../../prometheus/{0}.json".format(name)
```

Comment:
Shouldn't this use something from our `settings` instead?
```suggestion
    advertisement_path = os.path.join(settings.PROJECT_DIR, "prometheus", "{}.json".format(name))
```

(`pathlib.Path` might be nice, here.)

Reply:
Addressed in commit d5b010d731ad1ede03a5276af4b90d278b4994a5.